src/Http/Authentication.Abstractions/src/AuthenticateResult.cs

```diff
@@ -11,6 +11,8 @@ namespace Microsoft.AspNetCore.Authentication;
 /// </summary>
 public class AuthenticateResult
 {
+    private static readonly AuthenticateResult _noResult = new() { None = true };
```

Comment:
The only problem I see caching this is that some of the properties expose mutable objects. What happens if this is mutated?

Reply:
I see no problem with `_noResult`. All the properties are null and the setters are private.

https://github.com/dotnet/aspnetcore/blob/d9ee5d71e87a60bb766864ae5d7e0d81b9050c9e/src/Http/Authentication.Abstractions/src/AuthenticateResult.cs#L30-L45

Which properties for this case you think cause problem?

One problem might be that now `Clone` isn't really doing a deep copy for no result.

Reply:
I think you're right, I didn't notice the `protected set`.  I'm not sure clone should return the same instance though...